src/main/java/com/puppycrawl/tools/checkstyle/site/PropertiesMacro.java

```diff
@@ -307,7 +307,14 @@ private static void writeTokensList(Sink sink, List<String> tokens, String token
             writeLinkToToken(sink, tokenTypesLink, token);
         }
         sink.rawText(INDENT_LEVEL_18);
-        sink.text(SiteUtil.DOT);
+
+        if (tokens.isEmpty()) {
+            sink.text("empty");
```

Comment:
Can we test with `<code>empty</code>` ? To avoid diff

Reply:
@romani, are you asking to write a unit-test case for the new branch i introduced? any PR for reference? And, do we write any test cases for PropertiesMacro.java?

Reply:
@romani, do you want me to change the code to something like below:
```
if (tokens.isEmpty()) {
            sink.text("<code>empty</code> ?");
```

Reply:
Yes, I hope it will do no changes for respectful line in xdoc

Reply:
@romani , 
```
        if (tokens.isEmpty()) {
            sink.rawText("<code>");
            sink.text("empty");
            sink.rawText("</code>");
        }
```

with above changes, the xml looks somewhat like this for properties matching the earlier format. 
```
     <subsection name="Properties" id="Properties">
        <div class="wrapper">
          <table>
            <tr>
              <th>name</th>
              <th>description</th>
              <th>type</th>
              <th>default value</th>
              <th>since</th>
            </tr>
            <tr>
              <td>format</td>
              <td>Define the RegExp for illegal pattern.</td>
              <td><a href="../../property_types.html#Pattern">Pattern</a></td>
              <td><code>&quot;^$&quot;</code></td>
              <td>3.2</td>
            </tr>
            <tr>
              <td>ignoreCase</td>
              <td>Control whether to ignore case when matching.</td>
              <td><a href="../../property_types.html#boolean">boolean</a></td>
              <td><code>false</code></td>
              <td>3.2</td>
            </tr>
            <tr>
              <td>message</td>
              <td>Define the message which is used to notify about violations; if empty then the default message is used.</td>
              <td><a href="../../property_types.html#String">String</a></td>
              <td><code>&quot;&quot;</code></td>
              <td>3.2</td>
            </tr>
            <tr>
              <td>tokens</td>
              <td>tokens to check</td>
              <td>subset of tokens
                <a href="../../apidocs/com/puppycrawl/tools/checkstyle/api/TokenTypes.html#NUM_DOUBLE">
                    NUM_DOUBLE</a>
                , <a href="../../apidocs/com/puppycrawl/tools/checkstyle/api/TokenTypes.html#NUM_FLOAT">
                    NUM_FLOAT</a>
                , <a href="../../apidocs/com/puppycrawl/tools/checkstyle/api/TokenTypes.html#NUM_INT">
                    NUM_INT</a>
                , <a href="../../apidocs/com/puppycrawl/tools/checkstyle/api/TokenTypes.html#NUM_LONG">
                    NUM_LONG</a>
                , <a href="../../apidocs/com/puppycrawl/tools/checkstyle/api/TokenTypes.html#IDENT">
                    IDENT</a>
                , <a href="../../apidocs/com/puppycrawl/tools/checkstyle/api/TokenTypes.html#COMMENT_CONTENT">
                    COMMENT_CONTENT</a>
                , <a href="../../apidocs/com/puppycrawl/tools/checkstyle/api/TokenTypes.html#STRING_LITERAL">
                    STRING_LITERAL</a>
                , <a href="../../apidocs/com/puppycrawl/tools/checkstyle/api/TokenTypes.html#CHAR_LITERAL">
                    CHAR_LITERAL</a>
                , <a href="../../apidocs/com/puppycrawl/tools/checkstyle/api/TokenTypes.html#TEXT_BLOCK_CONTENT">
                    TEXT_BLOCK_CONTENT</a>
                  .
              </td>
              <td><code>empty</code></td>
              <td>3.2</td>
            </tr>
          </table>
        </div>
      </subsection>
```

But, i am getting an issue of 
`[ERROR] [checkstyle] [ERROR] /Users/piyushkumarsadangi/Development/new/checkstyle/src/main/java/com/puppycrawl/tools/checkstyle/site/PropertiesMacro.java:311:26: The String "<code>" appears 2 times in the file. [MultipleStringLiterals]`

It is being originally used here in `PropertiesMacro` in below:
```
private static void writePropertyDefaultValueCell(Sink sink, String propertyName,
                                                      Field field, Object instance)
            throws MacroExecutionException {
        sink.rawText(INDENT_LEVEL_14);
        sink.tableCell();

        if (SiteUtil.TOKENS.equals(propertyName)) {
            final AbstractCheck check = (AbstractCheck) instance;
            if (check.getRequiredTokens().length == 0
                    && Arrays.equals(check.getDefaultTokens(), TokenUtil.getAllTokenIds())) {
                sink.text(SiteUtil.TOKEN_TYPES);
            }
            else {
                final List<String> configurableTokens = SiteUtil
                        .getDifference(check.getDefaultTokens(),
                                check.getRequiredTokens())
                        .stream()
                        .map(TokenUtil::getTokenName)
                        .collect(Collectors.toList());
                writeTokensList(sink, configurableTokens, SiteUtil.PATH_TO_TOKEN_TYPES);
            }
        }
        else if (SiteUtil.JAVADOC_TOKENS.equals(propertyName)) {
            final AbstractJavadocCheck check = (AbstractJavadocCheck) instance;
            final List<String> configurableTokens = SiteUtil
                    .getDifference(check.getDefaultJavadocTokens(),
                            check.getRequiredJavadocTokens())
                    .stream()
                    .map(JavadocUtil::getTokenName)
                    .collect(Collectors.toList());
            writeTokensList(sink, configurableTokens, SiteUtil.PATH_TO_JAVADOC_TOKEN_TYPES);
        }
        else {
            final String defaultValue = SiteUtil.getDefaultValue(
                    propertyName, field, instance, currentModuleName);
            System.out.println("default value: " + defaultValue);
            sink.rawText("<code>");
            sink.text(defaultValue);
            sink.rawText("</code>");
        }

        sink.tableCell_();
    }
```

Can i alter the check to allow multiple strings?

Reply:
> Yes, I hope it will do no changes for respectful line in xdoc

if i do this, 
```
if (tokens.isEmpty()) {
            sink.text("<code>empty</code>");
```
the test cases in xdocs fail. Also, the xml generated doesn't sync with original xml type. Hence, made changes as per the convention followed in `PropertiesMacro`. But, it is giving me multple strings issue.

Reply:
Move such string values to class constants to avoid violations.

Reply:
Update: I have created constants for the same to avoid the check violation.

Reply:
> Move such string values to class constants to avoid violations.

done

Reply:
@romani , is it good enough to be merged?